eventually/authentication/views.py

```diff
@@ -5,16 +5,109 @@
+        Handles GET request, retrieves a user profile from the database
+        :param request: request from the web page
+        :type request: WSGIRequest
+
+        :param user_id: id of a user to return.
+        :type user_id: int
+
+        :return: user dictionary
+            E.G.
+                {
+                "id": 24,
+                "first_name": "TechGG76",
+                "middle_name": "Postgres",
+                "last_name": "Jr.",
+                "email": "postman424@co.uk",
+                "created_at": 1509831239,
+                "updated_at": 1509835437
+                }
+        """
+
+        user = CustomUser.get_by_id(user_id=user_id)
+        if user:
+            user = user.to_dict()
+            return JsonResponse(user, status=200)
+        return HttpResponse('User does not exist', status=404)
+
+
+    def put(self, request, user_id):
+        """
+        Handles PUT request, modifies the user profile in the datatbase.
+
+        :param request: request from the web page with a json containing changes to be applied
+        :type request: WSGIRequest
+        :Example: incoming JSON request:
+        | {
+        |   "email":"pss@com.ua",
+        |   "last_name":"Ariana"
+        | }
+
+        :param user_id: id of a user which has to be changed
+        :type user_id: int
+
+        :return: status 200 if user has been updated,
+                 status 400 if user hasn't been updated,
+        """
+
+        new_attrs = json_loads(request.body)
+        if new_attrs:
+            user_to_modify = CustomUser.get_by_id(user_id)
+
+            if new_attrs.get('password') and password_validator(new_attrs['password']):
+                new_password = new_attrs['password']
+            else:
+                return HttpResponse("Password doesn't meet the requirements", status=400)
+
+            if new_attrs.get('email') and email_validator(new_attrs['email']):
+                new_email = new_attrs['email']
+            else:
+                return HttpResponse("You did not provide a valid email", status=400)
+
+            try:
+                user_to_modify.update(first_name=new_attrs.get('first_name'),
+                                      last_name=new_attrs.get('last_name'),
+                                      middle_name=new_attrs.get('middle_name'),
+                                      email=new_email,
+                                      password=new_password)
+                return HttpResponse('User has been updated', status=200)
+            except IntegrityError:
+                pass
+            return HttpResponse(status=400)
+
+        return HttpResponse(status=400)
+
+
+    def delete(self, request, user_id):
+        #pylint: disable=unused-argument
+
+        """
+        Handles delete request
+
+        :param user_id: id of a user to delete
+        :type user_id: int
+
+        :return: status 200
+        """
+
+        if CustomUser.delete_by_id(user_id):
```

Comment:
with that logic, I can simply delete your account =)

Reply:
Now you can't.)